core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -1350,13 +1349,20 @@ public DefaultWeightingFactory(EncodingManager encodingManager, GraphHopperStora
             this.ghStorage = ghStorage;
         }
 
-        public Weighting createWeighting(ProfileConfig profile, PMap hints) {
+        public Weighting createWeighting(ProfileConfig profile, PMap requestHints) {
+            // Merge profile hints with request hints, the request hints take precedence.
+            // Note that so far we do not check if overwriting the profile hints actually works with the preparation
+            // for LM/CH. Later we should also limit the number of parameters that can be used to modify the profile.
+            PMap hints = new PMap();
+            hints.putAll(profile.getHints());
+            hints.putAll(requestHints);
```

Comment:
These are (the very simple) rules how we merge the profile hints with the request hints. The only thing I am wondering now is how we do this for block_area? Shall we do this in a similar way (use hints from profile and overwrite with request hints)?

Reply:
> The only thing I am wondering now is how we do this for block_area?

Hmmh, we would need to add it to the list of shapes

> Do we need this here or is it ok to only create block areas from request hints (not the profile) here

I would postpone this and properly support this in CustomWeighting.

> Now it looks all we are missing to do this in createWeighting are the points

Yeah, although we only need this because of the is-in-area check which we could keep out of the Weighting creation somehow.